services/spar/src/Spar/API.hs

```diff
@@ -248,25 +250,49 @@ idpDelete zusr idpid = withDebugLog "idpDelete" (const Nothing) $ do
     Data.deleteIdPConfig idpid issuer team
     Data.deleteIdPRawMetadata idpid
   return NoContent
+  where
+    updateOldIssuers :: IdP -> Spar ()
+    updateOldIssuers _ = pure ()
+    -- we *could* update @idp ^. SAML.idpExtraInfo . wiReplacedBy@ to not keep the idp about
+    -- to be deleted in its old issuers list, but it's tricky to avoid race conditions, and
+    -- there is little to be gained here: we only use old issuers to find users that have not
+    -- been migrated yet, and if an old user points to a deleted idp, it just means that we
+    -- won't find any users to migrate.  still, doesn't hurt mucht to look either.  so we
+    -- leave old issuers dangling for now.
+
+    updateReplacingIdP :: IdP -> Spar ()
+    updateReplacingIdP idp = forM_ (idp ^. SAML.idpExtraInfo . wiOldIssuers) $ \oldIssuer -> do
+      wrapMonadClient $ do
+        iid <- Data.getIdPIdByIssuer oldIssuer
+        mapM_ (Data.clearReplacedBy . Data.Replaced) iid
```

Comment:
same comment here as for `setReplacedBy`: This can insert a record into the idp table with only an idp id and no other fields, in certain cases, leading to inconsistent data. Can we guard against that? Perhaps add a comment to this end as well to remind someone in the future refactoring this code and using these queries.

Reply:
The idp exists in line 266.  There is a possibility that it gets removed between 266 and 267, but that requires two admins working against each other on the same task, *and* a lot of luck.

But I like your suggestions to make the cql code more robust, and I'll look into them!